src/lib/shadow-shim-helper-rs/src/syscall_types.rs

```diff
@@ -249,3 +249,61 @@ impl std::fmt::Debug for SysCallReg {
             .finish()
     }
 }
+
+// implement conversions from `SysCallReg`
+
+impl TryFrom<SysCallReg> for nix::fcntl::OFlag {
+    type Error = ();
```

Comment:
> I think this change wouldn't be needed if TryFromSyscallReg were made non-pub or pub-crate?

No, I don't think that helps (and tried it to be sure). 

I think removing the blanket trait implemntation would fix it:

```
impl<T: TryFrom<SysCallReg>> TryFromSyscallReg for T {
    fn try_from_reg(reg: SysCallReg) -> Option<Self> {
        Self::try_from(reg).ok()
    }
}
```

but then we'd need to explicitly implement `TryFromSyscallReg` for the types that were covered by that before. That wouldn't be too bad, but OTOH I'm not sure it's worth the extra boilerplate vs getting rid of the indirection and using `TryFrom` directly. I can't think of cases where we'd want a `TryFromSyscallReg<T>` and specifically *not* want a `TryFrom<T>` that does the same conversion, but maybe I'm not thinking of something?

Reply:
Okay my thinking was that the compiler error would be because we had a blanket implementation, but the trait was also public so another crate like `shadow-shim-helper` could also provide its own conflicting implementation. And if we made the trait non-pub, then other crates couldn't provide their own implementation of the trait and there couldn't be any conflict with the blanket implementation, so the compiler would be happy. But it sounds like this is wrong.

The original reason for this `TryFromSyscallReg` was to be able to convert from `SysCallReg` to other types without polluting the actual `TryFrom<SysCallReg>` trait with these conversions that are only needed by the strace logger. For example it isn't obvious if `impl TryFrom<SysCallReg> for nix::fcntl::OFlag` is good to have generally everywhere in Shadow since it implicitly converts `SysCallReg` to an `i32` (or maybe a `u32`, I'm not sure), and at the time I wanted to narrow the scope of this to just the strace logger.

But anyways I'm good with any change, including how it is now.

Reply:
Ok - just added a comment in `TryFromSyscallReg` for now